src/Symfony/Component/DependencyInjection/ServiceLocator.php

```diff
@@ -82,6 +83,13 @@ public function count(): int
         return \count($this->getProvidedServices());
     }
 
+    public function getIterator(): \Traversable
+    {
+        foreach (array_keys($this->getProvidedServices()) as $id) {
+            yield $id => $this->get($id);
+        }
```

Comment:
This basically means that by iterating over the locator you initialize all of its services. To me, this kinda defeats the purpose of a service locator: I could've just injected a tagged iterator and called `iterator_to_array()` on it.

_If_ we really want to make the locator traversable (not sure if we should yet), maybe we'd rather yield a service closure instead?

Reply:
but because this is a generator, this still does it kinda lazily so it's OK?

Reply:
I don't know, this still looks like a pitfall to me. I wouldn't do it.

Reply:
@kbond WDYT of this concern? I don't feel the need for this extra safety myself but maybe your experience could help best decide?

Reply:
> I don't know, this still looks like a pitfall to me. I wouldn't do it.

Does the new interface alleviate the concern at all?